test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusProdModeTest.java

```diff
@@ -408,6 +454,32 @@ private void ensureApplicationStartupOrFailure() throws IOException {
         }
     }
 
+    @Override
+    public void interceptBeforeAllMethod(Invocation<Void> invocation, ReflectiveInvocationContext<Method> invocationContext,
+            ExtensionContext extensionContext) throws Throwable {
+        doIntercept(invocation);
+    }
+
+    @Override
+    public void interceptBeforeEachMethod(Invocation<Void> invocation, ReflectiveInvocationContext<Method> invocationContext,
+            ExtensionContext extensionContext) throws Throwable {
+        doIntercept(invocation);
+    }
+
+    @Override
+    public void interceptTestMethod(Invocation<Void> invocation, ReflectiveInvocationContext<Method> invocationContext,
+            ExtensionContext extensionContext) throws Throwable {
+        doIntercept(invocation);
+    }
```

Comment:
Do we need to also intercept `AfterAll` / `AfterEach`? And all the other methods :-)

Reply:
There might will be situations were we have to, but they are unlikely to ever come up with this kind of internal test (hopefully :laughing:). 
I say we leave it as is for now and fix if it ever comes up.